projects/plugins/jetpack/class.jetpack.php

```diff
@@ -3467,11 +3470,18 @@ public static function disconnect( $update_activated_state = true ) {
 		// let's make sure to not disconnect the production site.
 		$connection->disconnect_site( ! Identity_Crisis::validate_sync_error_idc_option() );
 
-		Identity_Crisis::clear_all_idc_options();
-
 		if ( $update_activated_state ) {
```

Comment:
@mdbitz This is the only part that worries me. If the site gets disconnected without explicitly calling `Jetpack::disconnect`, eg via the endpoint, this option will never be updated. Any suggestions?

Reply:
from what i understand the activated option is only used by Jetpack so it should stay in Jetpack maybe use the jetpack_site_disconnected hook?

Reply:
actually it would make sense to move it into jetpack_site_disconnected as that is tied to the hook.

Reply:
I've updated this PR with the following suggested approach:
Since we can't use the `$update_activated_state` param when disconnection is not happening via `Jetpack::disconnect`, I introduced a filter: `jetpack_update_activated_state_on_disconnect`, set to `true` by default.

This will only be `false` if disconnect is called from `plugin_deactivation`. Seems that this is the only place that we used to set the `$update_activated_state` param to `false`.

Given the above, I believe it's also safe to remove that param completely.